File: backend/schemas/job.py
from typing import Optional
from datetime import datetime
from pydantic import BaseModel

class StoryJobBase(BaseModel):
    theme: str
    packs: list[str]

class StoryJobResponse(BaseModel):
    job_id: str
    status: str
    created_at: datetime
    challenge_id: Optional[int] = None
    completed_at: Optional[datetime] = None
    error: Optional[str] = None

    class Config:
        from_attributes = True

File: frontend/src/types/job.ts
import { type Status } from "../types/status";

export type JobState = {
  jobId: string | null;
  setJobId: (jobId: string) => void;
};

export type JobResponse = {
  job_id: string;
  status: Status;
  challenge_id: number | null;
  error?: string;
};
